### ISSUE TITLE
PeftAdapterMixin.load_adapter takes the model out of eval()

### ISSUE THREAD

**Issue by njbrake** [huggingface/transformers#34469]
https://github.com/huggingface/transformers/blob/c2820c94916e34baf4486accae74760972183a2f/src/transformers/integrations/peft.py#L71

When you load a trained adapter using `load_adapter` it takes the model out of eval mode: I would expect this behavior from the `add_adapter` method, but not the loading method. 

```
model_dir = "/path/to/seq2seq/model"
tokenizer = AutoTokenizer.from_pretrained(model_dir)
model:PeftAdapterMixin = AutoModelForSeq2SeqLM.from_pretrained(model_dir).to(device)
model.eval()
for module in model.modules():
    assert module.training == False # This will not throw an exception
model.load_adapter(adapter_path)
for module in model.modules():
    assert module.training == False # This will throw an exception
```

**Comment by BenjaminBossan** [comment]
Thanks for reporting, I agree that it is not the desired behavior that the LoRA weights would be trainable after calling `model.load_adapter`. In fact, when using `PeftModel`, the LoRA weights are not trainable by default, so this behavior is also inconsistent.

@njbrake For now, you can manually call `model.eval()` after loading the adapter. I'll work on a PR to fix the behavior.

@Rocketknight1 Sourab (pacman) is no longer at HF, so no need to ping him :)

**Comment by BenjaminBossan** [comment]
@njbrake Could you please check if `any(p.requires_grad for p in model.parameters())`? I think it should return `False`, which means we're basically good here.

Of course, it would still be better if we didn't have `module.training is True` for the LoRA modules, but since those are just linear layers, it doesn't change their behavior.

**Comment by njbrake** [comment]
@BenjaminBossan thanks for the response, I'll check that!

I am pretty sure that this is an issue, because if I remember correctly, the LoRA module can have a dropout layer so this would have an effect on that. 

I was observing non-deterministic behavior when generating with beam search (which should be deterministic), until I re set the model.eval() after loading the adapters, so I don't think that everything is ok as-is.

**Comment by BenjaminBossan** [comment]
> I am pretty sure that this is an issue, because if I remember correctly, the LoRA module can have a dropout layer so this would have an effect on that.

Right, if dropout > 0, this is indeed an issue. I created a PR that should fix hat: #34509. Also, note that if you load the model via PEFT, you don't run into this issue:

```python
from peft import PeftModel
base_model = ...
peft_model = PeftModel.from_pretrained(base_model, <path-to-adapter>)
```
